addons/l10n_th/data/res.country.state.csv

```diff
@@ -0,0 +1,78 @@
+"id","country_id:id","name","code"
```

Comment:
I'm not sure the best approach for this one, I just thought since this is particularly for Thai, it'd be best to put it on the Thai localization module instead